hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -473,26 +473,25 @@ case class HoodieResolveReferences(sparkSession: SparkSession) extends Rule[Logi
           "version expression is not supported for time travel")
       }
 
-      val tableIdentifier = sparkAdapter.getCatalystPlanUtils.toTableIdentifier(plan)
```

Comment:
Alright, so after taking a deeper look this seems to be a piece of code that is problematic -- `toTableIdentifier` assumes that the name is 2 name-parts, when it can be 3.

What i think we need to do is simply make sure we handle it properly here. I don't believe we need to change anything around how the catalogs/tables are resolved actually.

Reply:
This changes can't change how the catalogs/tables are resolved, the `LogicalPlan` is not changed due to this pr. Just let HUDI internal can correctly determine whether an `UnresolvedRelation` is hudi table.